st2common/st2common/models/api/pack.py

```diff
@@ -117,27 +126,46 @@ class PackAPI(BaseAPI):
                 'type': 'object',
                 'description': 'Specification for the system components and packages '
                                'required for the pack.',
-                'properties': {
-                    'stackstorm': {
-                        'type': 'string',
-                        'description': 'Required StackStorm version. Examples: ">1.6", '
-                                       '">=1.8dev, <2.0"',
-                        'pattern': ST2_VERSION_REGEX,
-                    }
-                }
+                'default': {}
             }
         }
     }
 
     @classmethod
     def to_model(cls, pack):
-        parameters = pack.__dict__
-        parameters['keywords'] = parameters.get('keywords', [])
-        parameters['files'] = parameters.get('files', [])
-        parameters['dependencies'] = parameters.get('dependencies', [])
-        parameters['system'] = parameters.get('system', {})
-
-        model = cls.model(**parameters)
+        ref = pack.ref
+        name = pack.name
+        description = pack.description
+        keywords = getattr(pack, 'keywords', [])
+        version = str(pack.version)
+
+        # Note: If some version values are not explicitly surrounded by quotes they are recognized
+        # as numbers so we cast them to string
+        if getattr(pack, 'version', None):
+            version = str(pack.version)
+
+        # Special case for old version which didn't follow semver format (e.g. 0.1, 1.0, etc.)
+        # In case the version doesn't match that format, we simply append ".0" to the end (e.g.
+        # 0.1 -> 0.1.0, 1.0, -> 1.0.0, etc.)
```

Comment:
I'm still against this change. There should be validation in place, and incorrect version should be an error. Otherwise nobody will start using correct versioning, because "I can't be bothered with this" or "I'm too busy" or "it's working fine now" or whatever else people come up with, and whenever we deprecate this behavior, people will complain. Since we're making a few more breaking changes in 2.1, the best time to introduce string versioning is now: as a user you'll probably have to restructure your packs anyway, might as well spend 15 more seconds to append ".0" to your pack version.